src/main/cpp/src/json_parser.cuh

```diff
@@ -624,14 +675,20 @@ class json_parser {
     char const* to_match_str_pos,
     char const* const to_match_str_end,
     char* copy_destination,
-    write_style w_style)
+    escape_style w_style)
   {
```

Comment:
Extract a `match_unescaped_string` from this function? Since `try_parse_quoted_string` is doing 2 things.
`match_unescaped_string` matches a valid string and is used by `match_current_field_name`

Reply:
done